providers/themeProvider.tsx

```diff
@@ -1,9 +1,35 @@
-import { ThemeProvider as NextThemeProvider } from 'next-themes';
+import { ThemeProvider as NextThemeProvider, useTheme } from 'next-themes';
 import type { FC, PropsWithChildren } from 'react';
+import { useEffect } from 'react';
+
+const ThemeButtonListener = () => {
```

Comment:
Why is this a component? This should clearly be a hook.

Reply:
Also this shouldn't be needed, `ThemeProvider` already handles the storage, state change, and other things.

What you want is to probably change the button to switch the theme.

Reply:
https://github.com/pacocoursey/next-themes?tab=readme-ov-file#usetheme this is what you want to do.

Also, hooking into `document` APIs within React is discouraged as it breaks the "sandbox".